util/gen_log_guide.ts

```diff
@@ -0,0 +1,503 @@
+        '39|2021-04-26T14:12:35.4830000-04:00|10FF0001|Tini Poutini|159475|191948|10000|10000|0|0|-648.3234|-804.5252|8.570148|1.010669||e34562c0c4e117e3709cf28a0c7759f6',
+        '39|2021-04-26T14:12:38.5160000-04:00|10FF0001|Tini Poutini|178669|191948|10000|10000|0|0|-648.3234|-804.5252|8.570148|1.010669||7ebe348673aa2a11e4036274becabc81',
+        '39|2021-04-26T14:13:21.6370000-04:00|10592642|Senor Esteban|54792|54792|10000|10000|0|0|100.268|114.22|-1.837917E-09|3.141593||883da0db11a9c950eefdbcbc50e86eca',
+        '39|2021-04-26T14:13:21.6370000-04:00|106F5D49|O\'ndanya Voupin|79075|79075|10000|10000|0|0|99.93127|114.2443|-1.862645E-09|-3.141593||8ed73ee57c4ab7159628584e2f4d5243',
+      ],
+    },
+  },
+} as const;
+
+type LogGuideOptions = {
+  lang?: string;
+  type?: string;
+};
+
+const isLineType = (type?: string): type is LineDocTypes => {
+  return type !== undefined && type in lineDocs;
+};
+
+const mappedLogLines: LangObject<LineDocTypes[]> = {
+  en: [],
+};
+
+const config: markdownMagic.Configuration = {
+  outputDir: path.posix.relative(curPath, path.posix.join(curPath, 'docs')),
+  transforms: {
+    logLines(_content, options: LogGuideOptions): string {
+      const language = options.lang;
+      const lineType = options.type;
+      if (!isLang(language)) {
+        console.error(`Received invalid lang specification: ${language ?? 'undefined'}`);
+        process.exit(-1);
+      }
+      if (!isLineType(lineType)) {
+        console.error(`Received invalid type specification: ${lineType ?? 'undefined'}`);
+        process.exit(-2);
+      }
+
+      const lineDoc = lineDocs[lineType];
+
+      if (!lineDoc) {
+        console.error(`Received missing type specification: ${lineType ?? 'undefined'}`);
+        process.exit(-3);
+      }
+
+      mappedLogLines[language] ??= [];
+      mappedLogLines[language]?.push(lineType);
+
+      const logRepo = new LogRepository();
+      let ret = '';
+      const lineDef = logDefinitions[lineType];
+      const structureNetworkArray = [
+        lineDef.type,
+        '2021-04-26T14:11:35.0000000-04:00',
+      ];
+      let lastIndex = 0;
+
+      for (const [name, index] of Object.entries(lineDef.fields)) {
+        if (['type', 'timestamp'].includes(name))
+          continue;
+        structureNetworkArray[index] = `[${name}]`;
+        lastIndex = Math.max(lastIndex, index);
+      }
+
+      for (let index = 2; index <= lastIndex; ++index)
+        structureNetworkArray[index] ??= '';
+
+      const structureNetwork = structureNetworkArray.join('|');
+      const structureLogLine = ParseLine.parse(logRepo, structureNetwork);
+      const structureLog = structureLogLine?.properCaseConvertedLine ??
+        structureLogLine?.convertedLine ?? '';
+
+      const examplesNetwork = lineDoc.examples[language]?.join('\n') ?? '';
+      const examplesLogLine = lineDoc.examples[language]?.map((e) => {
```

Comment:
I think this `[language]` lookup here should behave like cactbot translation functions, where it defaults to English if the language itself doesn't provide different examples.  I suspect in the vast majority of cases there will not need to be translated log lines.